### PR TITLE
Changes to reflect update status in guest cluster snapshot CR.

### PR DESCRIPTION
**Pull request by swgupta-vmw**
The snapshot ID in the guest cluster used to contain the corresponding
supervisor cluster snapshot CR name. This was used to watch for
supervisor snapshot CRs to get the upload status in the guest. This
was recently changed and the snapshot ID no longer points to the
supervisor cluser snapshot CR.

Added a new field in Snapshot-Status that has the name of the supervisor
snapshot CR in the guest. We use the guest parameters map passed to the
paravirt snapshot function to return the supervisor snapshot CR name.
The rest of the code to watch for update status remains the same.

Signed-off-by: Swati Gupta <swgupta@swgupta-a01.vmware.com>

### CODE REVIEW

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/169/

File: pkg/backupdriver/backup_driver_controller_base.go [hunk 1]
```diff
@@ -387,16 +385,11 @@ func (ctrl *backupDriverController) syncSnapshotByKey(key string) error {
 
 	// For guest clusters, add supervisor Snapshot to guest snapshot mapping if not already added
 	// The key is <supervisor snapshot CR name> and value is <guest snapshot namespace>:<guest snapshot CR name>
```
Comment (deepakkinni) [inline pkg/backupdriver/backup_driver_controller_base.go:387]: Update comment

Comment (swgupta-vmw) [reply pkg/backupdriver/backup_driver_controller_base.go:387]: The comment still holds true. We are adding the svcSnapshotCR name as the key with the value being the guestNS:guestSnapshotCR

**Comment by svcbot-qecnsdp** [comment]
Congratulations !! 

All the pre-check tests passed
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/171/

**Comment by xing-yang** [comment]
lgtm

**Pull request merged**